README.md

```diff
@@ -78,3 +80,36 @@ poetry run pytest .
 # or 
 poetry run pytest <path-to-specific-test-file>
 ```
+
+## Running the App in Docker
+In this project you can build and run a docker image for both Production (using gunicorn to run the server) and Development (using flask for the server and allowing for dynamic debugging of the app).
+
+### Production
+#### Building the image
+Has the basic main dependencies for the project
+``` bash 
+docker build --target production <path/to/Dockerfile> --tag todo_app:prod
+```
+#### Running the container
+For running the Production image, you will need to pass your populated ```.env``` file containing your secrets through the ```docker run``` command:
+
+``` bash
+docker run --env-file ./.env -p 8000:5000 todo_app:prod
+```
+The gunicorn application will Run in port ```8000``` inside the container, so we can use ```-p``` to expose <container_port>:<host_port> to forward the app into our host port 5000 
+
+### Development
+#### Building the image
+Has the main project dependencies as well as the development & testing dependencies installed
+```bash
+docker build --target development <path/to/Dockerfile> --tag todo_app:dev 
+```
+#### Running the container
+To take advantage of flask's development server that allows for dynamic hot reloading of code changes, we will use ```--mount``` to mount the app files into the container. This way the developer can perform changes in the local files, whilst still running the server in the container.
+
+Once again you will still need to pass your populated ```.env``` file containing your secrets through the ```docker run``` command:
+
+```bash 
+docker run --env-file ./.env -p 5000:5000 --mount type=bind,source=<full/path/to/app/files/todo_app>,target=/DevOps-Course-Starter/todo_app todo_app:dev
```

Comment:
You should hopefully find that we can use the below syntax to save people having to customise it (I _believe_ it should work on Unix/Windows)
```suggestion
docker run --env-file ./.env -p 5000:5000 --mount type=bind,source="$(pwd)"/todo_app,target=/DevOps-Course-Starter/todo_app todo_app:dev
```

Reply:
Hmm interesting, it worked for me in Windows and WSL - sounds like it might be [a change in a newer version of PowerShell](https://stackoverflow.com/questions/76230805/docker-bare-in-non-quoted-field); ignore for now and I'll have to go away and parse that properly some time!

Reply:
For the record, I'm assured that wrapping the entire section in quotes rather than just the command should work in PowerShell >7.3 as well as other shells:
```
docker run --env-file ./.env -p 5000:5000 --mount "type=bind,source=$(pwd)/todo_app,target=/DevOps-Course-Starter/todo_app" todo_app:dev
```

I haven't tested myself, and not super important, but in case you're curious